tcp.c

```diff
@@ -537,14 +542,13 @@ tcp_connect(char *server)
 void
 tcp_disconnect(void)
 {
-	if (g_ssl)
-	{
-		if (!g_network_error)
-			(void) SSL_shutdown(g_ssl);
-		SSL_free(g_ssl);
-		g_ssl = NULL;
-		SSL_CTX_free(g_ssl_ctx);
-		g_ssl_ctx = NULL;
+	if (g_ssl_initialized) {
+		//gnutls_bye(g_tls_session, GNUTLS_SHUT_RDWR);
```

Comment:
Because when connecting to some of my test windows machines the connection hangs after calling this one and I had cheated by commenting it out as I had no powers left to debug this issue.
As  I have said above it this a showstopper I can debug and fix this issue.